configure fails

I checked out a fresh GHC and tried to build from scratch using `--configure=--enable-tarballs-autodownload`. It failed with.

    configure: error: in `C:/Neil/ghc':
    configure: error: C compiler cannot create executables
    See `config.log' for more details

I then ran `./configure --enable-tarballs-autodownload` by hand and it said:

    configure: Checking for Windows toolchain tarballs...
    checking for gcc... C:/Neil/ghc/inplace/mingw/bin/gcc.exe
    checking whether the C compiler works... no
    configure: error: in `/c/Neil/ghc':
    configure: error: C compiler cannot create executables
    See `config.log' for more details

In neither case did `config.log` say anything interesting. I cleaned everything and reran `boot`/`configure` by hand and it worked. I then ran `shake-build/build.bat -j` and it said:

    Configuration file shake-build/cfg/system.config is missing.
    Run the configure script either manually or via the build system by passing --configure[=ARGS] flag.`

That seems to be no longer true, as I've already run `configure`. I then passed `--configure=--enable-tarballs-autodownload` and it now seems to be working (at least it's got a lot further). 32bit Windows.

--- a/src/Environment.hs
+++ b/src/Environment.hs
@@ -11,6 +11,11 @@ setupEnvironment = do
     -- ghc-cabal refuses to work when GHC_PACKAGE_PATH is set (e.g. by Stack)
     unsetEnv "GHC_PACKAGE_PATH"
 
+    -- in MinGW if PWD is set to a Windows "C:\\" style path then configure
+    -- `pwd` will return the Windows path, and then modifying $PATH will fail.
+    -- See https://github.com/snowleopard/shaking-up-ghc/issues/189 for details.
+    unsetEnv "PWD"
+
     -- On Windows, some path variables start a prefix like "C:\\" which may
     -- lead to failures of scripts such as autoreconf. One particular variable
     -- which causes issues is ACLOCAL_PATH. At the moment we simply reset it

